### PR TITLE
Add v1.23.15-hotfix.20230114 binaries to VHD.

### PR DESCRIPTION
**Pull request by yizhang4321**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add v1.23.15-hotfix.20230114 binaries to VHD.

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by AbelHu** [comment]
It seems like that all Linux VHD build tests failed.

File: parts/linux/cloud-init/artifacts/manifest.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
             "1.22.15",
             "1.23.8-hotfix.20220620",
             "1.23.12",
-            "1.23.15",
+            "1.23.15-v1.23.15-hotfix.20230114",
```
Comment (ganeshkumar5699) [inline parts/linux/cloud-init/artifacts/manifest.json:47]: typo which led to the VHD failures: `1.23.15` repeated twice

Comment (yizhang4321) [reply parts/linux/cloud-init/artifacts/manifest.json]: Fixed now

**Review by ganeshkumar5699 — APPROVED** [review]

**Comment by ganeshkumar5699** [comment]
Branch also needs to be updated

**Pull request merged**